Sources/Client/ScoreboardView.cpp

```diff
@@ -297,5 +313,73 @@ namespace spades {
 				}
 			}
 		}
+
+		void ScoreboardView::DrawSpectators(float top,
+			float centerX) {
+			IFont *bigFont = client->bigTextFont;
+			IFont *font = client->textFont;
+			char buf[256];
+			std::vector<ScoreboardEntry> entries;
+
+			static const auto xPixelSpectatorOffset = 20.f;
+
+			int numSpectators = 0;
+			float totalPixelWidth = 0;
+			for (int i = 0; i < world->GetNumPlayerSlots(); i++) {
+				Player *p = world->GetPlayer(i);
+				if (!p) continue;
+				if (p->GetTeamId() != spectatorTeamId)
+					continue;
+
+				ScoreboardEntry ent;
+				ent.name = p->GetName();
+				ent.id = i;
+				entries.push_back(ent);
+
+				numSpectators++;
+
+				// Measure total width in pixels so that we can center align all the spectators
+				sprintf(buf, "#%d", ent.id);
+				totalPixelWidth += font->Measure(buf).x + font->Measure(ent.name).x + xPixelSpectatorOffset;
+			}
+			switch (numSpectators) {
+				case 0: return;
+				case 1:	strcpy(buf, "Spectator:");
```

Comment:
"Spectator" doesn't belong in the localization lists though, it's not translated in any language.